router/customdestinationmanager/customdestinationmanager_test.go

```diff
@@ -197,3 +269,126 @@ func TestKVManagerInvocations(t *testing.T) {
+			config := map[string]interface{}{
+				"useJSONModule": true,
+				"address":       redisAddr,
+				"db":            0,
+				"clusterMode":   false,
+			}
+			kvMgr := kvstoremanager.NewRedisManager(config)
+			db := kvMgr.GetClient()
+
+			stCd, er := customManager.send(event, kvMgr, config)
+			if er != "" {
+				t.Logf("Error: %s\n", er)
+				require.Contains(t, er, tc.expectedSendDataResponse.err)
+				return
+			}
+			require.Equal(t, tc.expectedSendDataResponse.statusCode, stCd)
+
+			msgMap := gjson.GetBytes(event, "message.value").String()
+			key := gjson.GetBytes(event, "message.key").String()
+			res, err := db.JSONGet(context.TODO(), key).Result()
+			require.NoError(t, err)
+			require.Equal(t, msgMap, res)
+		})
+	}
+}
+
+func TestRedisMgrForMultipleJSONsSameKey(t *testing.T) {
+	initCustomerManager()
+	customManager := New("REDIS", Opts{}).(*CustomManagerT)
+	someDestination := backendconfig.DestinationT{
+		ID: "someDestinationID1",
+		DestinationDefinition: backendconfig.DestinationDefinitionT{
+			Name: "REDIS",
+		},
+	}
+	err := customManager.onNewDestination(someDestination)
+	assert.Nil(t, err)
+
+	transformedResponse := transformedResponseJSON{
+		Message: map[string]interface{}{
+			"key":  "user:myuser-id",
+			"path": "mode-1",
+			"value": map[string]interface{}{
+				"key": "someKey",
+				"fields": map[string]interface{}{
+					"field1": "value1",
+					"field2": 2,
+				},
+			},
+		},
+		UserId: "myuser-id",
+	}
+
+	t.Run("When JSON module is loaded into Redis, path is mentioned and key is not present in Redis, should set into redis and fetching the data should be successful", func(t *testing.T) {
+		redisAddr, destroy := testutils.StartRedis(t, "redis/redis-stack-server", "latest")
+		defer destroy()
+		event, err := json.Marshal(transformedResponse)
+		require.NoError(t, err)
+		config := map[string]interface{}{
+			"useJSONModule": true,
+			"address":       redisAddr,
+			"db":            0,
+			"clusterMode":   false,
+		}
+		kvMgr := kvstoremanager.NewRedisManager(config)
+		db := kvMgr.GetClient()
+
+		stCd, _ := customManager.send(event, kvMgr, config)
+		require.Equal(t, http.StatusOK, stCd)
+		v, err := db.JSONGet(context.TODO(), "user:myuser-id", "$.mode-1").Result()
+		require.NoError(t, err)
+		require.JSONEq(t, `[{"key":"someKey","fields":{"field1":"value1","field2":2}}]`, v)
+	})
+
+	t.Run("When JSON module is loaded into Redis, path is mentioned and key is present in Redis, should set into redis and fetching the data should be successful", func(t *testing.T) {
+		redisAddr, destroy := testutils.StartRedis(t, "redis/redis-stack-server", "latest")
+		defer destroy()
+		event, err := json.Marshal(transformedResponse)
+		require.NoError(t, err)
+		config := map[string]interface{}{
+			"useJSONModule": true,
+			"address":       redisAddr,
+			"db":            0,
+			"clusterMode":   false,
+		}
+		kvMgr := kvstoremanager.NewRedisManager(config)
+		db := kvMgr.GetClient()
+		ctx := context.TODO()
```

Comment:
`context.TODO` is used to mark a contenxt that you are planning to replace in the future with a different one (e.g. one with a timeout). Is that what you are planning TODO (pun intended)? :stuck_out_tongue_winking_eye:

Reply:
We can think that way 😉 . On a serious note, Empty context is the intention here 🙂

Reply:
`context.Background()` it is then :+1: :slightly_smiling_face:

Reply:
Done